Azure.ResourceManager.Core.Tests/ResourceTests.cs

```diff
@@ -0,0 +1,74 @@
+        }
+
+        [TestCase(0, "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.ClassicStorage/storageAccounts/account1",
+            "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.ClassicStorage/storageAccounts/account1")]
+        [TestCase(0, "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.ClassicStorage/storageAccounts/account1",
+            "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.classicStorage/storageAccounts/account1")]
+        [TestCase(-1, "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.ClassicStorage/storageAccounts/account1",
+            "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.DiffSpace/storageAccounts/account2")]
+        [TestCase(1, "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.DiffSpace/storageAccounts/account1",
+            "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.ClassicStorage/storageAccounts/account2")]
+        [TestCase(0, "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.${?>._`/storageAccounts/account1",
+            "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.${?>._`/storageAccounts/account1")]
+        [TestCase(-1, "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.ClassicStorage/storageAccounts/${?>._`",
+            "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.ClassicStorage/storageAccounts/account2")]
+        public void CompareToString(int expected, string id1, string id2)
+        {
+            ResourceTest resource1 = new ResourceTest(id1);
+            Assert.AreEqual(expected, resource1.CompareTo(id2));
+        }
+
+        [Test]
+        public void CompareToNull()
+        {
+            ResourceTest resource1 = new ResourceTest("/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.ClassicStorage/storageAccounts/account1");
+            ResourceTest resource2 = null;
+            Assert.AreEqual(1, resource1.CompareTo(resource2));
+            Assert.AreEqual(1, resource1.CompareTo((string)null));
+        }
+
+        [Test]
+        public void CompareToSame()
+        {
+            ResourceTest resource1 = new ResourceTest("/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.ClassicStorage/storageAccounts/account1");
+            ResourceTest resource2 = resource1;
+            Assert.AreEqual(0, resource1.CompareTo(resource2));
+        }
+    }
+
+    public class ResourceTest : Resource
```

Comment:
Move this to its own file in a new folder called Resource inside the tes tproject.

Rename it to TestResource